File: questao1/c/01/main.c
#include <stdio.h>

int main(void) {
  char *s = "abcabcbb";
  int max = 0;

  for (char *start = s; *start != '\0'; start++) {
    int count = 0;
    char seen[256] = {0}; // tabela ASCII

    for (char *end = start; *end != '\0'; end++) {

      if (seen[(unsigned char)*end])
        break; // caractere repetido encontrado

      seen[(unsigned char)*end] = 1;
      count++;
    }

    if (count > max)
      max = count;
  }

  printf("Maior substring sem repetição: %d caracteres\n", max);
  return 0;
}

File: questao1/lua/01/main.lua
-- main.lua
local s = "abcdabcbbbbbbbbbbbbbabc"
local max = 0

for start = 1, #s do
    local count = 0
    local seen = {} -- tabela de caracteres vistos

    for i = start, #s do
        local ch = s:sub(i, i)

        if seen[ch] then
            print("ch: " .. ch)
            print("seen[" .. string.byte(ch) .. "] = " .. tostring(seen[ch] and 1 or 0))
            break
        end

        seen[ch] = true
        count = count + 1
    end

    if count > max then
        max = count
    end
end

print("Maior substring sem repetição: " .. max .. " caracteres")
